tests/t500-operator.cu

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+// *****************************************************************************
+extern "C" __global__ void setup(void *ctx, CeedInt Q, Fields_Cuda fields) {
+  const CeedScalar *weight = fields.inputs[0], *dxdX = fields.inputs[1];
```

Comment:
This is another step away from having a single approach to QFunctions that each backend adapts to their own needs. Currently we have three ways that users need to write their inputs to QFunctions (and in three different source files for the exact same operator/qfunction combo)

CPU: array of pointers to scalar array, as per the spec (will eventually be in separate header files)
OCCA: pair of scalar arrays with all inputs/outputs and an array of offsets
CUDA: structs of input/output arrays

I think OCCA and CUDA should be able to have a single approach that also works well for CPU and doesn't require much/any efforts on the part of the user. If there is a performance barrier to providing an array of device arrays, perhaps a common struct of arrays would allow us to unify these aproaches.

I want to make it as easy as possible for users to use libCEED with multiple backends on hetrogeneous architecture without having to do a lot of work maintaining multiple copies of idential logic for their problem.

@jedbrown, @camierjs thoughts on QFunctions inputs/outputs as arrays of pointers to arrays vs single arrays with an array of offsets vs a struct of arrays?